### DISCUSSION TITLE
create global cookie jar after asyncio.run

### DISCUSSION THREAD

**Discussion by RFYoung** [aio-libs/aiohttp#7397] [Q&A] [answered]
Hello! I am new to async programming in Python, and I have encountered a problem.

I want to run a async function and then create a global cookie jar for different sessions. However, there would be a `RuntimeError` as there is no event loop on main thread. Here is an minimal example.

```python
import asyncio
import aiohttp

async def func_1():
    print("Just a test.")

class CoookieJarWrapper:
    _cookie_jar: aiohttp.CookieJar

    def __init__(self, jar):
        self._cookie_jar = jar

    def get_jar(self):
        return self._cookie_jar


asyncio.run(func_1())
wrapper = CoookieJarWrapper(jar=aiohttp.CookieJar())
```

I need to both execute the async function and create a cookie jar. I am wondering if there are any best practices on that. Thx  :pray: .

**Comment by Dreamsorcerer** [comment, marked as answer]
Typically, you run your code in a main() function or similar:
```
async def main():
    await func_1()
    wrapper = ...
    ...

asyncio.run(main())
```
